mule-extensions-api/src/main/java/org/mule/runtime/extension/api/options/OptionsResolverContext.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.options;
+
+import static java.util.Optional.ofNullable;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class OptionsResolverContext<Connection, Configuration> {
+
+  private final Map<String, Object> requiredParameters;
+  private final Connection connection;
+  private final Configuration configuration;
+
+  public OptionsResolverContext(Map<String, Object> requiredParameters, Connection connection, Configuration configuration) {
+    this.requiredParameters = requiredParameters;
```

Comment:
on the contrary... seems to me like this class should be used a lot more instead of performing so many injections